prow/plugins/jira/jira_test.go

```diff
@@ -0,0 +1,233 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jira
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/andygrunwald/go-jira"
+	"github.com/google/go-cmp/cmp"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	jiraclient "k8s.io/test-infra/prow/jira"
+)
+
+type fakeJiraClient struct {
+	existingIssues []jira.Issue
+	existingLinks  map[string][]jira.RemoteLink
+	newLinks       []jira.RemoteLink
+}
+
+func (f *fakeJiraClient) GetIssue(id string) (*jira.Issue, error) {
+	for _, existingIssue := range f.existingIssues {
+		if existingIssue.ID == id {
+			return &existingIssue, nil
+		}
+	}
+	return nil, jiraclient.NewNotFoundError(fmt.Errorf("No issue %s found", id))
+}
+
+func (f *fakeJiraClient) GetRemoteLinks(id string) ([]jira.RemoteLink, error) {
+	return f.existingLinks[id], nil
+}
+
+func (f *fakeJiraClient) AddRemoteLink(id string, link *jira.RemoteLink) error {
+	if _, err := f.GetIssue(id); err != nil {
+		return err
+	}
+	f.newLinks = append(f.newLinks, *link)
+	return nil
+}
+
+const fakeJiraUrl = "https://my-jira.com"
+
+func (f *fakeJiraClient) JiraURL() string {
+	return fakeJiraUrl
+}
+
+type fakeGithubClient struct {
+	editedComments map[string]string
+}
+
+func (f *fakeGithubClient) EditComment(org, repo string, id int, body string) error {
+	if f.editedComments == nil {
+		f.editedComments = map[string]string{}
+	}
+	f.editedComments[fmt.Sprintf("%s/%s:%d", org, repo, id)] = body
+	return nil
+}
+
+func (f *fakeGithubClient) GetIssue(org, repo string, number int) (*github.Issue, error) {
+	return nil, nil
+}
+
+func (f *fakeGithubClient) EditIssue(org, repo string, number int, issue *github.Issue) (*github.Issue, error) {
+	return nil, nil
+}
+
+func TestHandle(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name                   string
+		event                  github.GenericCommentEvent
+		existingIssues         []jira.Issue
+		existingLinks          map[string][]jira.RemoteLink
+		expectedNewLinks       []jira.RemoteLink
+		expectedCommentUpdates map[string]string
+	}{
+		{
+			name: "No issue referenced, nothing to do",
+		},
+		{
+			name: "Link is created based on body",
+			event: github.GenericCommentEvent{
+				CommentID:  intPtr(1),
+				HTMLURL:    "https://github.com/org/repo/issues/3",
+				IssueTitle: "Some issue",
+				Body:       "Some text and also ABC-123",
+				Repo:       github.Repo{FullName: "org/repo"},
+				Number:     3,
+			},
+			existingIssues: []jira.Issue{{ID: "ABC-123"}},
+			expectedNewLinks: []jira.RemoteLink{{Object: &jira.RemoteLinkObject{
+				URL:   "https://github.com/org/repo/issues/3",
+				Title: "org/repo#3: Some issue",
+				Icon: &jira.RemoteLinkIcon{
+					Url16x16: "https://github.com/favicon.ico",
+					Title:    "GitHub",
+				},
+			},
+			}},
+			expectedCommentUpdates: map[string]string{"/:1": "Some text and also [ABC-123](https://my-jira.com/browse/ABC-123)"},
```

Comment:
Why not specify org&repo to get sensible comments here?

Reply:
Done